components/layout/construct.rs

```diff
@@ -303,6 +303,7 @@ impl<'a> FlowConstructor<'a> {
             layout_data.remove_compositor_layers(self.layout_context
                                                      .shared
                                                      .constellation_chan
+                                                     .lock().unwrap()
```

Comment:
This will probably be too slow. Can you pass the mutex to `remove_compositor_layers` and take the lock only if necessary?